src/lib.rs

```diff
@@ -352,6 +352,93 @@ impl Filter {
     }
 }
 
+/// The possible input methods.
+pub enum InputMethod {
+    /// Read the input from the local Git history.
+    GitHistory,
+
+    /// Read the specified input file.
+    LogFile(String),
+
+    /// Read from `stdin`.
+    Stdin,
+}
+
+impl InputMethod {
+    /// Create a new instance from the given command line options.
+    pub fn parse(matches: &getopts::Matches) -> Option<Self> {
+        let mut method = 0;
+
+        if matches.opt_present("git") {
+            method |= 1
+        }
+
+        if matches.opt_present("input") {
+            method |= 2
+        }
+
+        if matches.opt_present("stdin") {
+            method |= 4
+        }
+
+        let method = method;
+
+        if method == 1 {
```

Comment:
Note: this could be done with a match statement.

Reply:
You reviewed the outdated version by accident.  The current version encapsulates the magic numbers in local constants to add further input methods in the future easier.

By the way:  I already tried to match against the local constants but it did not work as expected.  Suggestions?

Reply:
That is because they are not constants. You need to declare constants with the `const` keyword. You can do this also local.

Reply:
Thanks! :wrench:

Reply:
:wrench: :heavy_check_mark: